src/app/shell/provider/provider-activities/create-activity/create-address/create-address.component.ts

```diff
@@ -0,0 +1,87 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { Select, Store } from '@ngxs/store';
+import { Observable, Subject } from 'rxjs';
+import { debounceTime, distinctUntilChanged, startWith, takeUntil } from 'rxjs/operators';
+import { CityFilterService } from './../../../../../shared/services/filters-services/city-filter/city-filter.service';
+import { CityList } from './../../../../../shared/store/meta-data.actions';
+import { MetaDataState } from './../../../../../shared/store/meta-data.state';
+
+@Component({
+  selector: 'app-create-address',
+  templateUrl: './create-address.component.html',
+  styleUrls: ['./create-address.component.scss']
+})
+export class CreateAddressComponent implements OnInit {
+
+  AddressFormGroup: FormGroup;
+
+  @Output() passAddressFormGroup = new EventEmitter();
+  cities: string[] = [];
+  noCity: boolean=false;
+  destroy$: Subject<boolean> = new Subject<boolean>();
+  selectedCity:string;
+
+  @Select(MetaDataState.filteredCities)
+  filteredCities$: Observable<string[]>;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    public filterCityService: CityFilterService, 
+    public store: Store
+    ){
+    
+    this.AddressFormGroup = this.formBuilder.group({
+    street: new FormControl(''), 
+    buildingNumber: new FormControl(''),
+    city: new FormControl()
+    });
+  }
+
+  ngOnInit(): void {
+    this.passAddressFormGroup.emit(this.AddressFormGroup);
+    
+    this.filterCityService.fetchCities()
+      .subscribe((data)=>{
+        this.cities=data;
+    })
+    this.onInit();
+  }
+
+  private _filter(value: string): string[] {
+    const filterValue = value.toLowerCase();
+    let filteredCities = this.cities.filter(city => city.toLowerCase().startsWith(filterValue));
+    if(filteredCities.length===0){
+      this.noCity=true;
+      return ["Такого міста немає"]
```

Comment:
move this to state;
I can see that you are using strings as cities, but It is more likly objects, be proactive in designing this feature to avoid reworks in future